mpmath/functions/functions.py

```diff
@@ -164,11 +164,80 @@ def expm1(ctx, x):
 
 @defun_wrapped
 def log1p(ctx, x):
-    if not x:
-        return ctx.zero
-    if ctx.mag(x) < -ctx.prec:
-        return x - 0.5*x**2
-    return ctx.log(ctx.fadd(1, x, prec=2*ctx.prec))
+    LOG1P_EXTRAPREC = 10  # ctx._wrap_specfun()
+    # Note that all cases could by handled by log(1+c) provided the
+    # add is done exactly. Our aim here is be much faster than that,
```

Comment:
```suggestion
    # add is done exactly. Our aim here is to be much faster than that,
```
Adding a "to" missing from my original.